src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -1258,7 +1262,14 @@ private void ImportBinaryOperation(ILOpcode opcode)
                     else if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.ARM64)
                     {
                         _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.ThrowDivZero), "_divbyzero");
+                        if (opcode == ILOpcode.rem)
+                        {
+                            _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.Overflow), "_ovf");
+                        }
                     }
+
+                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.DblRem), "rem");
+                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.FltRem), "rem");
```

Comment:
I'm not sure I'm following how the logic flows here...

`x % y` will be the `rem` IL opcode which morph will then remap as a call to `CORINFO_HELP_DBLREM` or `CORINFO_HELP_FLTREM`.

* For `nativeaot/Runtime/MathHelpers` we still have an `fmod` helper that forwards down to `std::fmod`.
* For `ILCompiler.Compiler/Compiler/JitHelper` we then map these helpers to `Math.FMod`
* The `ILImporter.Scanner` here adds a dependency on `ReadyToRunHelper.DblRem/FltRem` for the `rem` IL opcode
* `vm/ecall.cpp` binds `CORINFO_HELP_DBLREM` to `Math.FMod`

But I don't see anything that actually takes `Math.FMod` and forwards it down to the CRT `fmod` API. It appeas to just remain an `Intrinsic, InternalCall`?

Reply:
> It appeas to just remain an Intrinsic, InternalCall?

Right. It depends on the pre-existing FMod FCalls: https://github.com/dotnet/runtime/blob/main/src/coreclr/classlibnative/float/floatdouble.cpp#L187-L191

Maybe the PR title is misleading. It is not a fully managed rewrite.

Reply:
Now that you have pointed this out, I have realized that https://github.com/dotnet/runtime/pull/100222/files#diff-0140426a976d6eaccae362fd5ec97d140649034a6d07199ed44b1b12738c007fR160-R166 is an expensive way to compute the FMod FCall pointers. We can just bake in the pointer directly.

Reply:
> Now that you have pointed this out, I have realized that https://github.com/dotnet/runtime/pull/100222/files#diff-0140426a976d6eaccae362fd5ec97d140649034a6d07199ed44b1b12738c007fR160-R166 is an expensive way to compute the FMod FCall pointers. We can just bake in the pointers directly.

I wanted to avoid special casing it for the FCall here since @tannergooding wanted to rewrite FMod into a full managed implementation later on.